src/integ_test_resources/ios/sdk/integration/cdk/cdk_integration_tests_ios/iot_stack.py

```diff
@@ -205,7 +213,78 @@ def setup_custom_authorizer(self):
             },
         )
 
-    def setup_custom_authorizer_function(self) -> str:
+    def setup_custom_authorizerUserPass(self):
+        custom_authorizer_name = "iot_custom_authorizerUserPass"
+        self._parameters_to_save["custom_authorizerUserPass_name"] = custom_authorizer_name
+        token_key_name = "IoTTokenKeyName"
+        self._parameters_to_save["custom_authorizerUserPass_token_key_name"] = token_key_name
+        token_value = "allow"
+        self._parameters_to_save["custom_authorizerUserPass_token_value"] = token_value
+
+        iot_custom_authorizer_key_resource = self.create_custom_authorizer_signing_key_generic("2",
+            "Manages an asymmetric CMK and token signature for iot custom authorizer with username and password.",
+            token_value)
+
+        custom_authorizer_token_signature = iot_custom_authorizer_key_resource.get_att(
+            "custom_authorizer_token_signature"
+        ).to_string()
+        self._parameters_to_save[
+            "custom_authorizerUserPass_token_signature"
+        ] = custom_authorizer_token_signature
+
+        #Force region to 'us-east-1' due to enhanced custom authorizers only available in this region
+        authorizer_function_arn = self.setup_custom_authorizer_function("2",
+                                                                        "custom_resources/iot_custom_authorizerUserPass_function",
+                                                                        "iot_custom_authorizerUserPass.handler",
+                                                                        "Sample custom authorizer that allows or denies based on username and password",
+                                                                        "us-east-1")
```

Comment:
Not super thrilled about putting us-east-1 here, but we are required to run this in us-east-1 -- any ideas?